### PR TITLE
Report the searched paths when the config file can't be found

### PR DESCRIPTION
**Pull request by 304NotModified**
Always wanted this ❤️ 

image

### CODE REVIEW

**Comment by snakefoot** [comment]
Great idea. Don't know how exception-safe the method `GetDefaultCandidateConfigFilePaths` is?

**Comment by 304NotModified** [comment]
Thanks!



> Don't know how exception-safe the method `GetDefaultCandidateConfigFilePaths` is?

Not sure due to the _appEnvironment stuff. But loading the config was already not (also documented) as non threadsafe? Good question by the way

**Comment by 304NotModified** [comment]
PS funny locations, not sure if this is expected


image

**Comment by snakefoot** [comment]
> But loading the config was already not (also documented) as non **threadsafe**?

I was thinking of exception-safe.

> PS funny locations, not sure if this is expected

Looks as expected to me (Checking both entry-assembly-location and process-location).

**Comment by 304NotModified** [comment]
> I was thinking of exception-safe.

oops :)

**Comment by 304NotModified** [comment]
Updated it to make it exception-safe :)

File: src/NLog/LogFactory.cs [hunk 1]
```diff
@@ -1165,7 +1165,7 @@ internal LogFactory LoadConfiguration(string configFile, bool optional)
 #if SILVERLIGHT
                     throw new System.IO.FileNotFoundException($"Failed to load NLog LoggingConfiguration from file {actualConfigFile}");
 #else
-                    var message = CreateFileNotFoundMessage(configFile);
+                    var message = CreateFileNotFoundMessage(_configLoader.GetDefaultCandidateConfigFilePaths(configFile));
```
Comment (snakefoot) [inline src/NLog/LogFactory.cs:1168]: Not sure I understand. Why is `_configLoader.GetDefaultCandidateConfigFilePaths` no inside the try-catch-block?

Comment (304NotModified) [reply src/NLog/LogFactory.cs:1168]: Well I prefer static methods, and `_configLoader.GetDefaultCandidateConfigFilePaths(configFile)` is lazy. But maybe it's a bit too much coincidence

Comment (304NotModified) [reply src/NLog/LogFactory.cs]: I was also doubting about this, so 2 hints = fix. :)

pushed!

**Comment by 304NotModified** [comment]
@snakefoot 

is maybe the text "The following locations were searched" better? (like MVC)

**Comment by snakefoot** [comment]
Think `Searched the following locations: {0}` is great. Like when the parameter-values comes right after parameter-name :)

But I think both sentences are very easy to understand.

**Pull request merged**